### ISSUE TITLE
zpool can't create pools bigger than 2 TB

### ISSUE THREAD

**Issue by n1mh** [openzfs/zfs#1741]
Hi!

I've found a problem when dealing with 4 TB disks in a server with linux kernel (3.2.43) and zfsonlinux (0.6.1). I'm trying making a pool with the whole disk and zpool only takes 1.98 TB.

```
# fdisk -l /dev/sdb

Disk /dev/sdb: 4398.0 GB, 4398046248960 bytes
255 heads, 63 sectors/track, 534698 cylinders
Units = cylinders of 16065 * 512 = 8225280 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk identifier: 0x00000000

Disk /dev/sdb doesn't contain a valid partition table
# zpool create pool1 sdb -f
# zpool list
NAME    SIZE  ALLOC   FREE    CAP  DEDUP  HEALTH  ALTROOT
pool1  1,98T   105K  1,98T     0%  1.00x  ONLINE  -
```

I've made a partition using parted and I've created a pool in it and it works fine. But, again, after a reboot, the pool is unavailable:

```
# parted /dev/sdc --script -- mklabel gpt
# parted /dev/sdc --script -- mkpart primary ext2 1 -1
# parted /dev/sdc --script -- print
Model: DELL PERC H710P (scsi)
Disk /dev/sdc: 4398GB
Sector size (logical/physical): 512B/512B
Partition Table: gpt

Numero  Inicio  Fin     Tamaño  Sistema de ficheros  Nombre   Banderas
 1      1049kB  4398GB  4398GB                       primary
# zpool create pool2 sdc1
# zpool list
NAME    SIZE  ALLOC   FREE    CAP  DEDUP  HEALTH  ALTROOT
pool1  1,98T   105K  1,98T     0%  1.00x  ONLINE  -
pool2  3,97T   105K  3,97T     0%  1.00x  ONLINE  -
# shutdown -r now
...
# zpool list
NAME    SIZE  ALLOC   FREE    CAP  DEDUP  HEALTH  ALTROOT
pool1  1,98T   105K  1,98T     0%  1.00x  ONLINE  -
pool2      -      -      -      -      -  FAULTED  -
```

I've found out this case, https://github.com/zfsonlinux/zfs/issues/1087, when Bodhisatva said something similar but much better explained. I've made sure that my RAID card has no limits creating a XFS pool with 4 TB without problems. This XFS pool works fine even after a reboot. And, for sure, it fails too adding a disk to an existing pool, if the result is bigger than 2 TB.

To sum up, I'm concern about this behaviour and I would like to know if zol version 0.6.2 solves this issue. Has anyone have the same problem?

Thanks,
diego
